workers/unity/Assets/Playground/Scripts/Cubes/ProcessColorChangeSystem.cs

```diff
@@ -25,26 +42,24 @@ protected override void OnUpdate()
                 var renderer = data.Renderers[i];
                 foreach (var colorEvent in component.Buffer)
                 {
-                    var color = UnityEngine.Color.white;
-                    switch (colorEvent.Payload.Color)
+                    if (materialPropertyBlocks.TryGetValue(colorEvent.Payload.Color, out var materialPropertyBlock))
```

Comment:
If we ever where to add a color later, this would fail silently.
`var materialPropertyBlock = materialPropertyBlocks[colorEvent.Payload.Color];` would ensure this throws an exception if you give a value it doesnt know about yet.